### DISCUSSION TITLE
Welcome to flutter-webrtc Discussions!

### DISCUSSION THREAD

**Discussion by cloudwebrtc** [flutter-webrtc/flutter-webrtc#428] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by kbramprasath** [comment]
android device  while recording need local stream video and both audio (input and output) .now its working local stream and mick audio only how ca n i achieve.please help me
As I tried to record the video call using webrtc library. Either i can record the initiator call audio or the receiver end audio.

public static void startRecordingToFile(String filePath, Integer id, @Nullable VideoTrack videoTrack, @Nullable AudioChannel audioChannel, EglBase eglBase, PeerConnectionConfClient peerConnectionClient, AudioSamplesInterceptor inputSamplesInterceptor) throws Exception {



    AudioSamplesInterceptor interceptor = null;
    if (audioChannel == AudioChannel.INPUT)
        interceptor = inputSamplesInterceptor;
    else if (audioChannel == AudioChannel.OUTPUT) {



        if (outputSamplesInterceptor == null)
            outputSamplesInterceptor = new OutputAudioSamplesInterceptor(peerConnectionClient.getJavaAudioDeviceModule());
        interceptor = outputSamplesInterceptor;
    }
    MediaRecorderImpl mediaRecorder = new MediaRecorderImpl(id, videoTrack, interceptor, eglBase);
    File file = new File(filePath);
    mediaRecorder.startRecording(file);
    mediaRecorders.append(id, mediaRecorder);
}

**Reply by lavishmanghnani2** [reply]
@kbramprasath did you get any solution to record audio and video ?

**Comment by aditdiqdpratama** [comment]
can i use this for SIP voice call?

**Reply by cloudwebrtc** [reply]
Here is flutter's sip plugin package, based on flutter-webrtc and sip over websockets
https://github.com/flutter-webrtc/dart-sip-ua

**Comment by piyushChopra20** [comment]
Can any one provide me an example or logic of having a multi peer connection with flutter webrtc so i can have a group calling feature in my app
i can handle two peers but above that not able to lay down the procedure for more than 2 peers in a call. can any one help?

**Reply by umarshaikh64** [reply]
please contact me

**Comment by laxkarparas1** [comment]
when i'm trying to record remote video. it throws Bad element exception.
can anyone provide me an example of mediaRecording for both local and remote

**Comment by shivanshtalwar0** [comment]
why flutter_webrtc do not work on safari? and how do you plan to tackle that?

**Comment by aramter** [comment]
How can I change the STUN and TURN servers to my own STUN and TURN?

**Reply by aramter** [reply]
Ok I found it
https://github.com/flutter-webrtc/dart-sip-ua/pull/93/commits/e5e335760ae24c904c6b901eef7f0c4840b5be0e
